File: back-end/src/main/java/com/huyphan/dtos/OrderItemDto.java
package com.huyphan.dtos;

import lombok.Getter;
import lombok.Setter;

/** Order item DTO. */
@Getter
@Setter
public class OrderItemDto extends ItemDto {
	/** Date in which this order item was placed. */
	private String placedIn;

	/** Uniquely identify an order. */
	private long id;

	/** Shipping address of this order item. */
	private ShippingAddressDto shippingAddress;
}

File: front-end/src/app/core/services/dtos/order-item-dto.ts
import { CartItemDto } from './cart-item-dto';
import { ShippingAddressDto } from './shipping-address-dto';

/** Order item DTO. */
export interface OrderItemDto extends CartItemDto {
  /** Date at which this item was placed.
   * @example 2007-07-25
   */
  readonly placedIn: string;

  /** Shipping address of this order item. */
  readonly shippingAddress: ShippingAddressDto;
}
